### PR TITLE
Dev/common tests

### PR DESCRIPTION
**Pull request by LukeWalker-CA**
* Add CaBoardUnitTest
  * Framework for CA unit tests using gtest
  * Includes OTP programming and USB printing common functions
  * Child classes have to write their own simTick function
* Added SerialStatus tests
  * Common tests for golden path, incorrect board, serial and status printouts that all boards should match

### CODE REVIEW

File: unit_testing/Util/serialStatus_tests.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
+/*!
+** @file   ll_tests.cpp
```
Comment (matias-cphatomics) [inline unit_testing/Util/serialStatus_tests.cpp:2]: Comments need to be updated

File: unit_testing/Util/serialStatus_tests.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+#include "serialStatus_tests.h"
+#include "pcbversion.h"
+
+using namespace std;
+
+/***************************************************************************************************
+** TESTS
+***************************************************************************************************/
+
+void goldenPathTest(SerialStatusTest& sst, const char* pass_string, int firstPrintTick) {
```
Comment (matias-cphatomics) [inline unit_testing/Util/serialStatus_tests.cpp:19]: I think all the general test functions should have clear comments stating their purpose :=)

File: unit_testing/Util/serialStatus_tests.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/*!
+** @file   caBoardsUnitTests.h
```
Comment (matias-cphatomics) [inline unit_testing/Util/serialStatus_tests.h:2]: wrong filename :)

File: unit_testing/Util/serialStatus_tests.h [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+** @author Luke W
+** @date   09/08/2024
+*/
+
+#pragma once
```
Comment (matias-cphatomics) [inline unit_testing/Util/serialStatus_tests.h:7]: Should we stick to the `#ifndef/#define` rather than `#pragma once`?

**Review by matias-cphatomics — APPROVED** [review]

File: unit_testing/Util/serialStatus_tests.h (hunk 4 above)
Comment (LukeWalker-CA) [reply unit_testing/Util/serialStatus_tests.h:7]: Yes - I was in a hurry when I was writing it, so used pragma, but I'd prefer to stick to #ifndef. I think its easier to manipulate.

**Pull request merged**